File: php/features.php
<?php
declare(strict_types=1);

// UuidGenerator SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class UuidGeneratorFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new UuidGeneratorBaseFeature();
            case "test":
                return new UuidGeneratorTestFeature();
            default:
                return new UuidGeneratorBaseFeature();
        }
    }
}

File: py/features.py
# UuidGenerator SDK feature factory

from feature.base_feature import UuidGeneratorBaseFeature
from feature.test_feature import UuidGeneratorTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UuidGeneratorBaseFeature(),
        "test": lambda: UuidGeneratorTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
